receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers.go

```diff
@@ -206,6 +210,92 @@ func (ghs *githubScraper) getPullRequests(
 	return pullRequests, nil
 }
 
+func (ghs *githubScraper) getCommitInfo(
+	client graphql.Client,
+	repoName string,
+	branch BranchNode,
+) (int, int, int64, error) {
+	comCount := 100
+	var cc *string
+	var adds int
+	var dels int
+	var age int64
+
+	// We're using BehindBy here because we're comparing against the target
+	// branch, which is the default branch. In essence the response is saying
+	// the default branch is behind the queried branch by X commits which is
+	// the number of commits made to the queried branch but not merged into
+	// the default branch. Doing it this way involves less queries because
+	// we don't have to know the queried branch name ahead of time.
+	comPages := getNumPages(float64(100), float64(branch.Compare.BehindBy))
+
+	for nPage := 1; nPage <= comPages; nPage++ {
+		if nPage == comPages {
+			comCount = branch.Compare.BehindBy % 100
+			// When the last page is full
+			if comCount == 0 {
+				comCount = 100
+			}
+		}
+		c, err := ghs.getCommitData(client, repoName, comCount, cc, branch.Name)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting commit data", zap.Error(err))
+			return 0, 0, 0, err
+		}
+
+		if len(c.Edges) == 0 {
+			break
+		}
+		cc = &c.PageInfo.EndCursor
+		if nPage == comPages {
+			e := c.GetEdges()
+			oldest := e[len(e)-1].Node.GetCommittedDate()
+			age = int64(time.Since(oldest).Seconds())
+		}
+		for b := 0; b < len(c.Edges); b++ {
+			adds = add(adds, c.Edges[b].Node.Additions)
+			dels = add(dels, c.Edges[b].Node.Deletions)
+		}
+
+	}
+	return adds, dels, age, nil
+}
+
+func (ghs *githubScraper) getCommitData(
+	client graphql.Client,
+	repoName string,
+	comCount int,
+	cc *string,
+	branchName string,
+) (*CommitNodeTargetCommitHistoryCommitHistoryConnection, error) {
+	data, err := getCommitData(context.Background(), client, repoName, ghs.cfg.GitHubOrg, 1, comCount, cc, branchName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(data.Repository.Refs.Nodes) == 0 {
+		return nil, errors.New("no commits returned")
```

Comment:
Is there a valid use case where a branch exists but doesn't have commits? Should that be an error?

Reply:
It's a possibility. I've updated the code/comments to reflect. Let me know your thoughts.

Reply:
Makes sense, thanks for explaining 👍